pixel-saver@deadalnix.me/decoration.js

```diff
@@ -48,7 +48,7 @@ function guessWindowXID(win) {
 	
 	// use xwininfo, take first child.
 	let act = win.get_compositor_private();
-	if (act) {
+	if (act && act['x-window']) {
```

Comment:
Ok now, we don't need to do the array lookup twice.

  let xwindow = act && act['x-window'];
  if (xwindow) {
    let xwininfo = GLib.spawn_command_line_sync('xwininfo -children -id 0x%x'.format(xwindow));
  }